.eslintrc.js

```diff
@@ -1,25 +1,29 @@
 module.exports = {
-    "env": {
+    root: true,
+    env: {
         "browser": true,
         "es6": true
     },
-    "extends": "eslint:recommended",
-    "parserOptions": {
+    extends: [
+    //    "plugin:@typescript-eslint/recommended",
+    //    "plugin:@typescript-eslint/recommended-requiring-type-checking",
+    ],
+    parser: '@typescript-eslint/parser',
+    parserOptions: {
         "ecmaVersion": 2020,
-        "sourceType": "module"
+        "sourceType": "module",
+        "project": "./tsconfig.json"
     },
-    "rules": {
-        "no-console": "off",
-        "no-empty": "off",
-        "no-prototype-builtins": "off",
-        "no-unused-vars": "warn"
-    },
-    "globals": {
-        "DEFINE_VERSION": "readonly",
-        "DEFINE_GLOBAL_HASH": "readonly",
-        // only available in sw.js
-        "DEFINE_UNHASHED_PRECACHED_ASSETS": "readonly",
-        "DEFINE_HASHED_PRECACHED_ASSETS": "readonly",
-        "DEFINE_HASHED_CACHED_ON_REQUEST_ASSETS": "readonly"
+    plugins: [
+        '@typescript-eslint',
+    ],
+    rules: {
+        "@typescript-eslint/no-floating-promises": 2,
+        "@typescript-eslint/no-misused-promises": 2,
+        "no-unused-vars": "off",
+        "@typescript-eslint/no-unused-vars": ["warn"],
+        "no-undef": "off",
+        "semi": ["error", "always"],
+        "@typescript-eslint/explicit-function-return-type": ["warn"]
```

Comment:
This is giving a lot of lint errors. I'd rather add this in another PR along with fixes for the errors.

Reply:
Actually this is supposed to be .ts-eslintrc.js, I just copy it over while developing because my editor refuses to acknowledge any other file. I'll remove that line for now anyways.

Reply:
https://github.com/vector-im/hydrogen-web/pull/851/commits/7fffdfb037cd353d92954a09dd1e9f3e574561e3